src/flamegpu/io/Telemetry.cpp

```diff
@@ -0,0 +1,167 @@
+#include "flamegpu/io/Telemetry.h"
+
+#include <cstdlib>
+#include <cerrno>
+#include <algorithm>
+#include <memory>
+#include <array>
+#include <iostream>
+#include <cstdio>
+#include <cctype>
+
+
+#include "flamegpu/version.h"
+#include "flamegpu/exception/FLAMEGPUException.h"
+#include "flamegpu/util/Environment.h"
+
+
+
+namespace flamegpu {
+namespace io {
+namespace Telemetry {
+
+const char TELEMETRY_ENDPOINT[] = "https://nom.telemetrydeck.com/v1/";
+const char TELEMETRY_APP_ID[] = "94AC5E3F-F674-4E29-BF87-DAF4BA7F8F79";
+
+std::string generateTelemetryData(std::string event_name, std::map<std::string, std::string> payload_items) {
+    const std::string var_testmode = "$TEST_MODE";
+    const std::string var_appID = "$APP_ID";
+    const std::string var_buildHash = "$BUILD_HASH";
+    const std::string var_eventName = "$EVENT_TYPE";
+    const std::string var_payload = "$PAYLOAD";
+
+    // check ENV for test variable FLAMEGPU_TEST_ENVIRONMENT
+    std::string testmode;
+    if (flamegpu::util::isTestEnvironment())
+        testmode = "true";
+    else
+        testmode = "false";
+    std::string appID = TELEMETRY_APP_ID;
+    std::string buildHash = flamegpu::BUILD_HASH;
+
+    // add app version by checking to see if python run
+    if (flamegpu::util::hasEnvironmentVariable("FLAMEGPU_PYFLAMEGPU_VERSION")) {
```

Comment:
Hadn't thought of this. Think I've always used swig flags in headers to detect if python (e.g. for changing splash screen for vis).

Reply:
I hadn't though to use a swig if def. I guess its ok to leave this.

Reply:
I'm not sure why we need to use env vars at all, they can be manually set by users which would potentialyl break things. 

Might as well use the macro (or add a constexpr at namespace scope using the macros value) as this should always be the same value for a given static library build (given we have diff methods exposed for swig or not). Though I supose a user could theoretically take the compiled-for-swig static library and link a c++ binary to it for some reason, at which point it would need some runtime detection too, but we don't care about that use-case.

Reply:
Switched to SWIG macro checking.